### PR TITLE
fix AContainer can't find recipes with similar items

### PR DESCRIPTION
**Pull request by lijinhong11**
## Description

#4166 

## Proposed changes

Skip scanned slots

## Related Issues (if applicable)

#4166 

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [X] I have made sure that the proposed changes do not break compatibility across the supported Minecraft versions (1.16.* - 1.20.*).
- [ ] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

**Comment by JustAHuman-xD** [comment]
Have you tested this?

**Comment by lijinhong11** [comment]
not :( but I will test later

**Comment by lijinhong11** [comment]
Tested and it did.

File: src/main/java/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/abstractItems/AContainer.java [hunk 1]
```diff
@@ -422,6 +422,8 @@ protected MachineRecipe findNextRecipe(BlockMenu inv) {
             for (ItemStack input : recipe.getInput()) {
                 for (int slot : getInputSlots()) {
                     if (SlimefunUtils.isItemSimilar(inventory.get(slot), input, true)) {
+                        if (found.containsKey(slot)) continue; //skip scanned slots
```
Comment (JustAHuman-xD) [inline src/main/java/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/abstractItems/AContainer.java:425]: You should invert this check and merge it with the surrounding of statement, have it be the first boolean expression

**Review by JustAHuman-xD — COMMENTED** [review]
Just a code style (and kinda performance but just micro optimization) thing

**Comment by lijinhong11** [comment]
So there's the only way is to stack items' amount

**Comment by JWJUN233233** [comment]
im afraid that it wont work.
if there is a recipe a[Diamondx64] and a recipe b[Diamondx64, Iron_ingot] and there are 64 diamonds in the AContainer, it should match recipe a instead of b. i may give another solution later.

**Comment by lijinhong11** [comment]
u r more excellent than me 🙁
________________________________
发件人: Ddggdd135 ***@***.***>
发送时间: 2024年4月27日 08:34
收件人: Slimefun/Slimefun4 ***@***.***>
抄送: lijinhong11 ***@***.***>; Author ***@***.***>
主题: Re: [Slimefun/Slimefun4] fix AContainer can't find recipes with similar items (PR #4172)


im afraid that it wont work.
if there is a recipe a[Diamondx64] and a recipe b[Diamondx64, Iron_ingot] and there are 64 diamonds in the AContainer, it should match recipe a instead of b. i may give another solution later.

**Comment by JWJUN233233** [comment]
and i wonder why not use hashcode. if the NBT of item is different from that in recipe, it shouldnt match.

**Comment by JustAHuman-xD** [comment]
> and i wonder why not use hashcode. if the NBT of item is different from that in recipe, it shouldnt match.

Because there are tons of cases where nbt wouldn't be identical but the ingredient is still valid

**Comment by JWJUN233233** [comment]
> > and i wonder why not use hashcode. if the NBT of item is different from that in recipe, it shouldnt match.
> 
> Because there are tons of cases where nbt wouldn't be identical but the ingredient is still valid 

Could you please give an example?

**Comment by JustAHuman-xD** [comment]
> Could you please give an example?

Sure! We already have problems with it in other areas so I can just give a couple of those: the recipe calls for a diamond hoe, you have a diamond hoe that was used and then repaired, nbt won't be identical.

2. You have a recipe that calls for fertilizer, the fertilizer has a custom model nbt because the server previously had a resource pack but now does not or the model number changed, nbt won't be identical

Etc etc

**Comment by JWJUN233233** [comment]
thats so bad

**Comment by JustAHuman-xD** [comment]
> thats so bad

?

**Comment by JWJUN233233** [comment]
i crrated my pr #4177

**Pull request closed without merging**